test/CadlRanchProjects.Tests/type-property-nullable.cs

```diff
@@ -285,5 +289,175 @@ public Task Type_Property_Nullable_CollectionsString_patchNull() => Test(async (
+        {
+            var requiredStringList = TypeAsserts.HasProperty(typeof(StringProperty), nameof(StringProperty.NullableProperty), BindingFlags.Public | BindingFlags.Instance);
+
+            Assert.Null(requiredStringList.SetMethod);
+        }
+
+        [Test]
+        public void NotRequiredNullableListsSerializedAsNull()
+        {
+            var inputModel = new CollectionsStringProperty();
+            inputModel.NullableProperty = null;
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Null, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void NotRequiredNullableListsSerializedEmptyWhenCleared()
+        {
+            var inputModel = new CollectionsStringProperty("required", new List<string> {"1", "2"});
+            inputModel.NullableProperty.Clear();
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Array, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void NotRequiredNullablePropertiesSerializeWhenSet()
+        {
+            var inputModel = new CollectionsStringProperty();
+            inputModel.RequiredProperty = "required";
+            inputModel.NullableProperty = new List<string> { "1", "2" };
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual("2", element.GetProperty("nullableProperty")[1].GetString());
+        }
+
+        [Test]
+        public void NotRequiredNullablePropertiesDeserializedWithNulls()
+        {
+            var model = CollectionsStringProperty.DeserializeCollectionsStringProperty(JsonDocument.Parse("{\"NonRequiredNullableString\": null}").RootElement);
+            Assert.Null(model.NullableProperty);
+        }
+
+        [Test]
+        public void NullablePropertiesSerializedAsNulls()
+        {
+            var inputModel = new CollectionsStringProperty("required", null);
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Null, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void NullablePropertiesSerializedAsEmptyLists()
+        {
+            var inputModel = new CollectionsStringProperty("required", Array.Empty<string>());
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Array, element.GetProperty("nullableProperty").ValueKind);
+            Assert.AreEqual(0, element.GetProperty("nullableProperty").GetArrayLength());
+        }
+
+        [Test]
+        public void NullablePropertiesDeserializedAsNullsWithUndefined()
+        {
+            var model = CollectionsStringProperty.DeserializeCollectionsStringProperty(JsonDocument.Parse("{}").RootElement);
+            Assert.IsNull(model.NullableProperty);
+        }
```

Comment:
if this is how our code is implemented, this is wrong.
Undefined should be deserialized as `undefined collection` (`new ChangeTrackingList<T>`). Only explicit `null` will deserialize into `null`.

Reply:
I will remove this test from this PR for now to avoid affecting its merge. I will create a separate PR to include this test.